src/libstd/num/f32.rs

```diff
@@ -120,18 +190,110 @@ impl Float for f32 {
-
+    /// Returns `true` if the number is `NaN`.
+    ///
+    /// ```
+    /// use std::f32;
+    ///
+    /// let nan: f32 = f32::NAN;
+    ///
+    /// assert!(nan.is_nan());
+    /// assert!(!7.0f32.is_nan());
+    /// ```
     #[inline]
     fn is_nan(self) -> bool { num::Float::is_nan(self) }
+    /// Returns `true` if the number is infinite.
+    ///
+    /// ```
+    /// use std::Float;
+    /// use std::f32;
+    ///
+    /// let inf: f32 = Float::infinity();
+    /// let neg_inf: f32 = Float::neg_infinity();
+    /// let nan: f32 = f32::NAN;
+    ///
+    /// assert!(!7.0f32.is_infinite());
+    /// assert!(!nan.is_infinite());
+    ///
+    /// assert!(inf.is_infinite());
+    /// assert!(neg_inf.is_infinite());
+    /// ```
     #[inline]
     fn is_infinite(self) -> bool { num::Float::is_infinite(self) }
+    /// Returns `true` is the number is finite.
+    ///
+    /// ```
+    /// use std::Float;
+    /// use std::f32;
+    ///
+    /// let inf: f32 = Float::infinity();
+    /// let neg_inf: f32 = Float::neg_infinity();
+    /// let nan: f32 = f32::NAN;
+    ///
+    /// assert!(7.0f32.is_finite());
+    ///
+    /// assert!(!nan.is_finite());
+    /// assert!(!inf.is_finite());
+    /// assert!(!neg_inf.is_finite());
+    /// ```
     #[inline]
     fn is_finite(self) -> bool { num::Float::is_finite(self) }
+    /// Returns `true` if the number is neither zero, infinite, subnormal, or NaN.
+    ///
+    /// ```
+    /// use std::num::Float;
+    /// use std::f32;
+    ///
+    /// let min: f32 = f32::MIN_POS_VALUE; // 1.17549435e-38f32
+    /// let max: f32 = f32::MAX_VALUE;
+    /// let lower_than_min: f32 = 1.0e-40;
+    ///
+    /// assert!(min.is_normal());
+    /// assert!(max.is_normal());
+    ///
+    /// assert!(!0f32.is_normal());
+    /// assert!(!f32::NAN.is_normal());
+    /// assert!(!f32::INFINITY.is_normal());
+    /// // Values between `0` and `min` are Subnormal.
+    /// assert!(!lower_than_min.is_normal());
+    /// ```
     #[inline]
     fn is_normal(self) -> bool { num::Float::is_normal(self) }
+    /// Returns the floating point category of the number. If only one property
+    /// is going to be tested, it is generally faster to use the specific
+    /// predicate instead.
+    ///
+    /// use std::num::{Float, FpCategory};
+    /// use std::f32;
+    ///
+    /// let num = 12.4f32;
+    /// let inf = f32::INFINITY;
+    ///
+    /// assert_eq!(num.classify(), FpCategory::Normal);
+    /// assert_eq!(inf.classify(), FpCategory::Infinite);
     #[inline]
     fn classify(self) -> FpCategory { num::Float::classify(self) }
-
+    /// Returns the mantissa, base 2 exponent, and sign as integers. The original
+    /// number can be recovered by `sign * mantissa * 2 ^ exponent`. The number
+    /// was encoded with some bizarre scheme because floats are complex as seen at
+    /// [IEEE floating point][IEEE-floating-point] resulting in these strange
+    /// components.
```

Comment:
This needs to be reworded. I was shown on IRC the gist of what this is doing but I'm not familiar with it and it's unexpected. A better comment from someone knowledgeable about this would be helpful.

It definitely needs a comment because I expected `2` to decompose to `+1 * 2 * 10^0` or `+1 * 2 * 2^0`, not this `+1 * 8388608 * 2 ^ -22`. A really good link would be real nice or at least a good one. I'm not sure mine is good because I didn't read it carefully.